woocommerce/woocommerce-ios
# [Mobile Payments] Update Stripe SDK to 3.1.0

## Description
This PR updates stripeTerminal pod to 3.1.0

## Changes 
All changes are detailed in a checklist here: pdfdoF-49a-p2: 

- Updates Stripe SDK to `3.1.0`
- `stripeId` is now Optional in order to account for offline payments. Since our implementation does not have that option we'll extend it and provide a default empty string if that's the case, just for conformance.
- Updates `Terminal.shared` singleton property names, and instance methods.
- Updates `UnderlyingError` cases
- Updates `DiscoveryConfiguration.init()`. Since this is now a protocol, we need to use the concrete configuration builders provided by Stripe.
- Updates `PaymentIntentParameters` with a builder, as above, since this is now immutable.
- Removes `CardDetails.fingerprint` and `CardPresentDetails.fingerprint`

There are two items in the SDK update that I believe do not need any change for the moment, but happy to iterate further. I described the rationale behind this here: pdfdoF-49a-p2#comment-5283

```
* `discoverReaders` is now completed when `connectReader` is called. This is a behaviour change from 2.x where `discoverReaders` would continue running until connect succeeded.
* Canceling `discoverReaders` now completes with an `SCPErrorCanceled` error. Previously no error was provided when canceled.
```

## Testing instructions

Test any combination of card reader connection, and payment flows that worked previously using card present or tap to pay. The same behaviour should be expected after the update. As a guide, on my testing I went through the following scenarios:

* Connection from Order payment flow
* Connection from Payments menu
* Connection of a reader with a mandatory update
* Connection of M2 reader
* Connection of WisePad 3 reader
* Connection from switched off state
* Take payment after in-line connection process
* Take payment with a previously-connected reader
* Take contactless payment
* Take card-inserted payment
* Reader update: error message when attempted with low battery
* Reader update: successfully updates software and starts payment (in that flow) when completed

I was not able to test the following, since I don't have the option:
* Connection of Chipper reader
* Take Interac payment
* Refund Interac payment



---
